File: #0387.first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {

        int length = s.length();
        unordered_map<char,int> m;
        for (int i = 0; i < length; i++) {
            m[s[i]]++;
        }

        for(int i = 0; i < length; i++) {
            if (m[s[i]] == 1) {
                return i;
            }
        }

        return -1;
    }
};

File: #0387.first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    const map = {};
    for (let c of s) {
        map[c] = (map[c] += 1) || 1;
    }
    
    for (let i = 0; i < s.length; i++) {
        if (map[s[i]] === 1) {
            return i;
        }
    }
    return -1;
};
